Pull request so we can discuss changes

erudnick-cohen opened the pull request.

pclalv commented on maps/LakeOfRage.asm:
```diff
@@ -180,9 +180,8 @@ Randomizer_RedGyaradosSpecies::
-	giveitem RED_SCALE
+	verbosegiveitem RED_SCALE
 	waitsfx
-	writetext LakeOfRageGotRedScaleText
```
let's delete the data at `LakeOfRageGotRedScaleText`, too.

erudnick-cohen replied:
For now I suggest keeping this in case its decided that this change should be locked behind a flag. I suspect that it probably won't need to be though, so once this gets properly merged into the main branch for speedchoice V7 we should propose to delete this.

pclalv commented on maps/GoldenrodFlowerShop.asm:
```diff
@@ -49,8 +47,6 @@ FlowerShopTeacherScript:
 	faceplayer
 	opentext
-	checkevent EVENT_FOUGHT_SUDOWOODO
-	iftrue .FoughtSudowoodo
```
let's delete the data associated with `.FoughtSudowoodo` - nothing references it anymore.

erudnick-cohen replied:
I know for a fact that dabomstew will be locking the sudowoodo related skipping stuff behind a flag (cinco bingo players may not want it turned on). Consequently, we shouldn't delete things associated with this since its likely this check will get added back in later on with additional flag related checking.

pclalv replied:
okay, so we can either
1. not delete this code, or
2. implement the flag.

pclalv replied:
deleting this doesn't make sense if stew's gonna touch it himself later. just let him do whatever with it.

pclalv commented on maps/MrPokemonsHouse.asm:
```diff
@@ -28,8 +28,7 @@ MrPokemonsHouse_MapScripts:
 	promptbutton
 	waitsfx
-	giveitem MYSTERY_EGG
-	writetext MrPokemonsHouse_GotEggText
```
let's delete `MrPokemonsHouse_GotEggText`.

erudnick-cohen replied:
For now I suggest keeping this in case its decided that this change should be locked behind a flag. I suspect that it probably won't need to be though, so once this gets properly merged into the main branch for speedchoice V7 we should propose to delete this.

pclalv replied:
same as the sudowoodo code above, then; i don't think this code should be deleted.

pclalv commented on maps/OlivinePort.asm:
```diff
@@ -397,7 +386,7 @@ OlivinePort_MapEvents:
 	db 7 ; object events
 	object_event  7, 23, SPRITE_SAILOR, SPRITEMOVEDATA_STANDING_UP, 0, 0, -1, -1, 0, OBJECTTYPE_SCRIPT, 0, OlivinePortSailorAtGangwayScript, EVENT_OLIVINE_PORT_SAILOR_AT_GANGWAY
-	object_event  7, 15, SPRITE_SAILOR, SPRITEMOVEDATA_STANDING_UP, 0, 0, -1, -1, 0, OBJECTTYPE_SCRIPT, 0, OlivinePortSailorBeforeHOFScript, EVENT_OLIVINE_PORT_SPRITES_BEFORE_HALL_OF_FAME
+	object_event  6, 15, SPRITE_SAILOR, SPRITEMOVEDATA_STANDING_UP, 0, 0, -1, -1, 0, OBJECTTYPE_SCRIPT, 0, OlivinePortSailorBeforeHOFScript, EVENT_OLIVINE_PORT_SPRITES_BEFORE_HALL_OF_FAME
```
what's this line about? does this just move the NPC blocking the gangway? if there's no point in keeping this NPC, then we might as well delete it and its associated data.

erudnick-cohen replied:
For now I suggest keeping the two versions of the npc in case its decided that this change should be locked behind a flag. I suspect that it probably won't need to be though, so once this gets properly merged into the main branch for speedchoice V7 we should propose to delete this.

pclalv replied:
okay, let's delete it now.

pclalv commented on maps/OlivinePort.asm:
```diff
@@ -82,12 +82,7 @@ OlivinePortWalkUpToShipScript:
 	checkevent EVENT_FAST_SHIP_FIRST_TIME
 	iffalse .FirstTime
-	readvar VAR_WEEKDAY
-	ifequal SUNDAY, .NextShipMonday
```
we can delete `.NextShipMonday` and `.NextShipFriday`.

erudnick-cohen replied:
For now I suggest keeping these in case its decided that this change should be locked behind a flag. I suspect that it probably won't need to be though, so once this gets properly merged into the main branch for speedchoice V7 we should propose to delete this.

pclalv commented on maps/VermilionPort.asm:
```diff
@@ -84,12 +84,6 @@ VermilionPortWalkUpToShipScript:
 	turnobject PLAYER, LEFT
 	opentext
-	readvar VAR_WEEKDAY
-	ifequal MONDAY, .NextShipWednesday
```
same here, might as well delete `.NextShipWednesday` and `.NextShipSunday`.

erudnick-cohen replied:
For now I suggest keeping these in case its decided that this change should be locked behind a flag. I suspect that it probably won't need to be though, so once this gets properly merged into the main branch for speedchoice V7 we should propose to delete this.

pclalv commented on maps/PokemonFanClub.asm:
```diff
@@ -77,10 +77,9 @@ PokemonFanClubClefairyGuyScript:
+	verbosegiveitem LOST_ITEM
 	iffalse .NoRoom
 	disappear POKEMONFANCLUB_FAIRY
-	writetext PokemonFanClubPlayerReceivedDollText
```
let's delete `PokemonFanClubPlayerReceivedDollText`.

erudnick-cohen replied:
For now I suggest keeping this in case its decided that this change should be locked behind a flag. I suspect that it probably won't need to be though, so once this gets properly merged into the main branch for speedchoice V7 we should propose to delete this.

pclalv commented on maps/RadioTower2F.asm:
```diff
@@ -100,8 +100,6 @@ TrainerGruntF2:
 	faceplayer
 	opentext
-	checkflag ENGINE_ROCKETS_IN_RADIO_TOWER
-	iftrue .MidRocketTakeover
```
let's delete `.MidRocketTakeover`

erudnick-cohen replied:
For now I suggest keeping these in case its decided that this change should be locked behind a flag. I suspect that it probably won't need to be though, so once this gets properly merged into the main branch for speedchoice V7 we should propose to delete this.

pclalv commented on maps/RadioTower5F.asm:
```diff
@@ -6,10 +6,11 @@
 	scene_script .DummyScene0 ; SCENE_DEFAULT
 	scene_script .DummyScene1 ; SCENE_RADIOTOWER5F_ROCKET_BOSS
 	scene_script .DummyScene2 ; SCENE_RADIOTOWER5F_NOTHING
+	scene_script .DummyScene3
```
i don't love that this adds a lot of duplicate code. i would prefer refactoring the whole thing so that the rocket boss is an NPC that you can talk to. it's up to you though.

erudnick-cohen replied:
Using scenes (triggers) is probably the easiest way I see to do this. It ensures that all NPCs maintain their vanilla behavior, which is less confusing for players. The duplication of code shouldn't really be much of a concern here (this IS assembly code after all).

pclalv replied:
that's fine.

pclalv commented on maps/BlackthornGym1F.asm:
```diff
@@ -29,7 +29,7 @@ BlackthornGym1F_MapScripts:
 BlackthornGymClairScript:
 	faceplayer
 	opentext
-	checkflag ENGINE_RISINGBADGE
```
it's not immediately clear what's going on here. i would prefer if we could make this clearer, or if we could delete this little conditional branch entirely.

erudnick-cohen replied:
Cleaned these up to not use the mt. silver flypoint. The only way to fail to receive the tms is to receive regular items instead (which only happens in full item rando). If this does occur, it will just give the badge a second time (which doesn't do anything and should be harmless).

pclalv replied:
right on, thanks.

pclalv commented on maps/GoldenrodUndergroundWarehouse.asm:
```diff
@@ -71,8 +71,8 @@ GoldenrodUndergroundWarehouseDirectorScript:
-	readvar VAR_BADGES
-	ifgreater 6, .GotCardKey
+	checkcode VAR_BADGES
+	ifless 7, .GotCardKey
```
this seems suspicious. what happens if the player gets 8 badges and then does the radio tower?

erudnick-cohen replied:
Then they receive the "card key" like they normally would. The comparison is used to check for when the player should NOT be receiving the card key (if its true, then it jumps to "GotCardKey" and skips the code that gives the card key).

pclalv reviewed (commented):
looks good. the only problem i see is that there's a lot of dead code that you've left behind - all of that should be cleaned up.

pclalv commented:
all of the "this will likely be flagged" changes are really weird. why are any of those changes in this branch if they're not what we're gonna be doing?

erudnick-cohen committed:
Item rando changes (#17)

* update gitignore

* Remove the rocket who can block the flower shop during the rocket takeover

* Flower shop gives item even after sudowoodo defeated

* make red scale verbosegiveitem and removed unneeded text box

* Switched mystery egg to verbosegiveitem and removed associated text box

* Force SS Aqua to operate every day of the week

* Revert "Force SS Aqua to operate every day of the week"

This reverts commit 7ec7151f9980b5bdaaf0a4dc32fe12c182401990.

* Revert "Switched mystery egg to verbosegiveitem and removed associated text box"

This reverts commit ad25e808dc09973fd74489bf755d07b358f37241.

* Revert "update gitignore"

This reverts commit 44546b6118472d24fd1b214423e6d4e456e7496f.

* Revert "Remove the rocket who can block the flower shop during the rocket takeover"

This reverts commit b80a0a177b05b4b1b44f04b5b6f0beef3a2de13c.

* Revert "make red scale verbosegiveitem and removed unneeded text box"

This reverts commit 2677390f674184c49c8b73a95e3cb5539a531e43.

* Revert "Flower shop gives item even after sudowoodo defeated"

This reverts commit d44f8f22b9275beca073cd401b4a0003bbb7b0f6.

* Make officer at lab optional to prevent scene change making pokeballs unobtainable

Officer can still be spoken too if you want to name the rival (or can just be skipped if you don't)

* Ensure player can still obtain secretpotion after giving it to jasmine

* Switched lost item to verbosegiveitem and removed assoicated text

* Make director permanently stuck in basement and have him check if team rocket is attacking before he gives out an item

Director will remain in basement after team rocket is defeated and his item can still be obtained at that point. However, his item cannot be obtained prior to the rocket invasion occuring.

* Radio card still available during rocket takeover

* Buena's item still available during rocket takeover

* Fight fake director or rocket boss in any order

NOTE: there are now two places in the code where basement key and clear bell are given out with this!

* Only need to talk to power plant manager once and machine part and zap cannon always available

Note: talking to the power plant manager a second time will respawn the rocket in cerulean gym, which will also respawn the machine part.

* Reroute blackthorn and goldenrod already beaten checks to mt silver flypoint

This ensures that you don't get locked out of these badges by obtaining the vanilla badge for these locations. Mt. silver flypoint is used because it requires access to all the badges in the first place.

* Switched kanto gym leaders to check if they are beaten, instead of checking if you have their vanilla badge

* Make dragon fang a more normal item so it doesn't cause harmless trippy graphics errors when it would be switched to verbosegiveitem

* Train can be taken without restoring power to kanto

Note that pass cannot normally be obtained unless you have already restored power... unless you're playing item rando.

* Before checking for everstone, elm needs to actually check if you have the egg

Otherwise it can get skipped, which then skips the pokeballs given by the aide

* Fixed power plant to require talking director once with no possibility of infinitely spawning rockets or machine parts

* Made director check actually work correctly

* Made dragon fang actually just a perfectly normal itemball

* Lost item doesn't require talking to copycat, so that you don't softlock by turning it in before checking it

* These changes should ONLY ever be used for testing

* Update .gitignore

* Fixed up blackthorn to use verbose give item and to have clair just check on giving you the tm to see if you have the badge already

Technically you can't fail to obtain the TM...except in full item randomizer (where this can happen since it might not be a TM). However, there is no harm in clair giving you a badge for a "second time" so this shouldn't really be an issue.

* Removed whitney's badge check so that it just checks if you recieved attract

This can technically fail in full item rando, but its otherwise harmless (getting a badge twice doesn't actually do anything).

* Revert "Merge branch 'master' into ItemRandoSupportTesting"

This reverts commit 5f35df1b0982dba2161309741957b74fe46cf3e3, reversing
changes made to 0a67dab01d35f9a58feffc31e98924bc3ec6490c.

* Changes from master for testing

* Try to make Ho-Oh chamber accesible by beating Ho-Oh. Also, make the rainbow wing obtainable by beating suicune

* Ho-Oh shouldn't appear if you haven't beaten the E4 yet

* Made ho oh chamber actually work and not randomly sometimes lock the door behind you

talking to the wall will always open the door

* Getting the vanilla rainbow wing requires defeating E4

* Comments in ho-oh chamber

* Tin tower comments

* Revert "These changes should ONLY ever be used for testing"

This reverts commit 0a67dab01d35f9a58feffc31e98924bc3ec6490c.

* Reinclude flower shop skip in changes

Dabomstew committed:
Bring back cop script, disable Violet Gym code that screws it up

Dabomstew committed:
Add item rando data block, make flower shop rocket gone if item rando on

Dabomstew committed:
Make sure flower shop rocket disappears

Dabomstew committed:
Put item rando basement director/flower shop/boat behavior behind flags

Dabomstew committed:
Tidy up changed item scripts

Dabomstew committed:
Clean up Radio Tower item rando flags/behavior

Dabomstew committed:
Cleanup syntax/comments

Dabomstew committed:
Tidy up Ho-Oh chamber, start on tin tower

Dabomstew committed:
Patch up tin tower 1F (for now, may need further adjustment)

Dabomstew committed:
Flag-lock early Lost Item

Dabomstew committed:
Make dragons den skip into its own option

Dabomstew committed:
Lock logic settings if item rando is active

Dabomstew committed:
Add support for progressive rods

Dabomstew committed:
Add support for badges to engine flag givers

Dabomstew committed:
Tidy up engine flag naming code

Dabomstew committed:
Use verbosesetflag for all gyms + rocket trigger

Dabomstew committed:
Remove redundant constant

Dabomstew committed:
Change secretpotion to verbosegiveitem

Dabomstew committed:
Use verbosesetflag for Pokegear, cards, Pokedex

Dabomstew committed:
Add option for early kanto dex access

Dabomstew committed:
Fix move menu acc display

Dabomstew committed:
Fix stat exp accumulation and move menu acc display

Dabomstew committed:
Fix CV generation

Dabomstew committed:
Fix box corruption when bill changes boxes before first save

Dabomstew committed:
v7.1

Dabomstew committed:
Fix non easy Clair script, make egg hatching respect nicknaming choice

erudnick-cohen committed:
Fixed elms lab so that it can't softlock items if the player obtains the mystery egg in place of the pokegear

erudnick-cohen committed:
Don't use verbosesetflag for the phone card

Currently not planing to use phone card for item rando, since its not an "item" in the original game in the first place.

erudnick-cohen committed:
Esnure bike also doesn't get locked by rockets

Dabomstew committed:
Merge pull request #21 from erudnick-cohen/master

Item rando related fixes

Dabomstew committed:
Remove bike music on save load if option toggled

Dabomstew committed:
Protect engine flag jingles (e.g. pokegear) from being overloaded

Dabomstew committed:
v7.2

vicendithas committed:
Prevent Clair from giving TM early when Easy Clair turned off

With Easy Clair off, Clair would give her TM if you talk to her after beating her, instead of forcing you to go to Dragon's Den and do the Shrine quiz to get it.

Dabomstew committed:
Merge pull request #24 from vicendithas/master

Prevent Clair from giving TM early when Easy Clair turned off - Fixes#23

erudnick-cohen committed:
Make it impossible to not get rocket passwords

Dabomstew committed:
Set result to true when we verbosesetflag

erudnick-cohen committed:
Add support for more ho-oh chamber related settings

erudnick-cohen committed:
add name entries to the item table for flags we actually use

erudnick-cohen committed:
Make leftovers a REGULAR item

erudnick-cohen committed:
pad red's text so that we can eventually make him say silly things

erudnick-cohen committed:
Merge branch 'Dabomstew:master' into master

erudnick-cohen committed:
Fix leftovers to work as a normal check

erudnick-cohen committed:
change where the tin related check scripts are

erudnick-cohen committed:
Fix text

erudnick-cohen committed:
Pad reds text

erudnick-cohen committed:
testing code

Dabomstew committed:
Fix verbosesetflag for real this time?

erudnick-cohen committed:
Merge pull request #4 from Dabomstew/master

Fix verbosesetflag for real this time?

erudnick-cohen committed:
Add support for unowndex

erudnick-cohen committed:
Merge pull request #5 from erudnick-cohen/Why-is-Rockets-suddenly-broken

Why is rockets suddenly broken

erudnick-cohen committed:
apparently this commit didn't go through?

erudnick-cohen committed:
Merge branch 'master' of https://github.com/erudnick-cohen/pokecrystal-speedchoice-new

erudnick-cohen committed:
v7.3

erudnick-cohen committed:
undo debug stuff

erudnick-cohen committed:
updated things

erudnick-cohen committed:
item rando doesn't start active

erudnick-cohen committed:
Make text not overflow

erudnick-cohen committed:
add badges to the names list

erudnick-cohen committed:
screw this, make it a verbosegiveitem, I don't understand how two different items were given there

erudnick-cohen committed:
aide less weird?

erudnick-cohen committed:
update ruins of alph to not have screwed over unown dex issues
